flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/SystemResourcesCounter.java

```diff
@@ -269,14 +269,14 @@ private void calculateNetworkUsage(List<NetworkIF> networkIFs) {
 
             receiveRatePerInterface.set(
                     i,
-                    (networkIF.getBytesRecv() - bytesReceivedPerInterface[i])
+                    (networkIF.getBytesRecv() - bytesReceivedPerInterface.get(i))
                             * 1000
                             / probeIntervalMs);
             sendRatePerInterface.set(
                     i,
                     (networkIF.getBytesSent() - bytesSentPerInterface[i]) * 1000 / probeIntervalMs);
 
-            bytesReceivedPerInterface[i] = networkIF.getBytesRecv();
+            bytesReceivedPerInterface.set(i, networkIF.getBytesRecv());
```

Comment:
if we want to be Thread safe should we not use [compareAndSet](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLongArray.html#compareAndSet-int-long-long-) rather than set?